csrc/scheduler/utils.cpp

```diff
@@ -915,6 +932,19 @@ PersistentBufferSizeReturn persistentBufferSize(
         std::max(max_proj_persistence_size, projected_buffer_size);
   }
 
+  // reduce max_persistence_size if a persistent buffer can be projected to
+  // other persistent tvs.
+  std::unordered_set<TensorView*> persistent_buffer_set(
```

Comment:
Persistent buffers are scoped, and the buffer size is the largest among multiple scopes. This reduction of the projection to immediate producers needs to be done with each scope.

Try to create an example with multiple scopes.

Reply:
Good point. In the newly added test case, the largest buffer size comes from a scope which doesn't have a projecatable buffer. The other scope has a projecatable buffer. So the previous implementation overestimated the buffer size. In the revised version, the `persistent_mask` is used in the buffer size calculation. It is set to `true` when 
```
    if (!canProjectToPersistentProducer(
            buffer, producers, persistent_buffer_set)) {
      persistent_mask[buffer_i] = true;
    }
```
when the mask is `false`, the size is not counted.